### PR TITLE
prune metric: accuracy 4/n

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

this template how to prune metrics and still hold compatibility for user
Followup for #6505

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,7 +65,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated `trainer.running_sanity_check` in favor of `trainer.sanity_checking` ([#4945](https://github.com/PyTorchLightning/pytorch-lightning/pull/4945))
 
 
-- Deprecated metrics in favor of `torchmetrics` ([#6505](https://github.com/PyTorchLightning/pytorch-lightning/pull/6505))
+- Deprecated metrics in favor of `torchmetrics` ([#6505](https://github.com/PyTorchLightning/pytorch-lightning/pull/6505),
+    [#6515](https://github.com/PyTorchLightning/pytorch-lightning/pull/6515))
```
Comment (Borda) [inline CHANGELOG.md:69]:
```suggestion


    [#6515](https://github.com/PyTorchLightning/pytorch-lightning/pull/6515),
    
    )
```

File: pytorch_lightning/metrics/classification/accuracy.py [hunk 2]
```diff
@@ -13,90 +13,16 @@
...
-        tensor(0.5000)
-
-        >>> target = torch.tensor([0, 1, 2])
-        >>> preds = torch.tensor([[0.1, 0.9, 0], [0.3, 0.1, 0.6], [0.2, 0.5, 0.3]])
-        >>> accuracy = Accuracy(top_k=2)
-        >>> accuracy(preds, target)
-        tensor(0.6667)
+    This implementation refers to :class:`~torchmetrics.Accuracy`.
 
+    .. warning:: This metric is deprecated, use ``torchmetrics.Accuracy``. Will be removed in v1.5.0.
```
Comment (SkafteNicki) [inline pytorch_lightning/metrics/classification/accuracy.py:25]:
```suggestion
    .. warning:: This metric is deprecated since v1.3.0, use ``torchmetrics.Accuracy``. Will be removed in v1.5.0.
```

File: pytorch_lightning/metrics/functional/accuracy.py [hunk 3]
```diff
@@ -55,68 +26,20 @@ def accuracy(
...
-        >>> accuracy(preds, target)
-        tensor(0.5000)
+    r"""
+    This implementation refers to :class:`~torchmetrics.functional.accuracy`.
 
-        >>> target = torch.tensor([0, 1, 2])
-        >>> preds = torch.tensor([[0.1, 0.9, 0], [0.3, 0.1, 0.6], [0.2, 0.5, 0.3]])
-        >>> accuracy(preds, target, top_k=2)
-        tensor(0.6667)
+    .. warning:: This metric is deprecated, use ``torchmetrics.functional.accuracy``. Will be removed in v1.5.0.
```
Comment (SkafteNicki) [inline pytorch_lightning/metrics/functional/accuracy.py:32]:
```suggestion
    .. warning:: This metric is deprecated since v1.3.0, use ``torchmetrics.functional.accuracy``. Will be removed in v1.5.0.
```

**Review by SkafteNicki — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -65,7 +65,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Deprecated `trainer.running_sanity_check` in favor of `trainer.sanity_checking` ([#4945](https://github.com/PyTorchLightning/pytorch-lightning/pull/4945))
 
 
-- Deprecated metrics in favor of `torchmetrics` ([#6505](https://github.com/PyTorchLightning/pytorch-lightning/pull/6505))
+- Deprecated metrics in favor of `torchmetrics` ([#6505](https://github.com/PyTorchLightning/pytorch-lightning/pull/6505),
+
+
+    [#6515](https://github.com/PyTorchLightning/pytorch-lightning/pull/6515),
+    
+    )
```
Comment (carmocca) [inline CHANGELOG.md:73]: this doesn't look right

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (Borda) [reply CHANGELOG.md:73]: it is just spacing around to avoid collisions

Comment (carmocca) [reply CHANGELOG.md:73]: But space is needed between bullet points (`-`), not within a bulletpoint

**Review by SeanNaren — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (Borda) [reply CHANGELOG.md:73]: ok, I'll check it with the next rebase as #6530 is required

**Comment by Borda** [comment]
ok, seems that the call once work very well, now for the tests purposes it needs to be reseted :D

File: tests/deprecated_api/test_remove_1-5_metrics.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 from pytorch_lightning.metrics.utils import get_num_classes, select_topk, to_categorical, to_onehot
 
 
-def test_v1_5_0_metrics_utils():
+def test_v1_5_metrics_utils():
     x = torch.tensor([1, 2, 3])
     with pytest.deprecated_call(match="It will be removed in v1.5.0"):
```
Comment (edgarriba) [inline tests/deprecated_api/test_remove_1-5_metrics.py:25]: I would expect this message to tell where is now placed the new function and how do I call it

**Review by edgarriba — APPROVED** [review]
LGTM. Nice tool @Borda

File: tests/deprecated_api/test_remove_1-5_metrics.py (hunk 5 above)
Comment (Borda) [reply tests/deprecated_api/test_remove_1-5_metrics.py:25]: yes it is there, which is just a fraction of the message :]

Comment (edgarriba) [reply tests/deprecated_api/test_remove_1-5_metrics.py:25]: awesome :)
